substreams/rpccalls.go

```diff
@@ -172,6 +187,128 @@ func (e *RPCEngine) ethCall(ctx context.Context, retryCount int, traceID string,
 	return cnt, deterministic, nil
 }
 
+func (e *RPCEngine) ETHGetBalance(ctx context.Context, traceID string, clock *pbsubstreams.Clock, in []byte) ([]byte, error) {
+	out, _, err := e.ethGetBalance(ctx, -1, traceID, clock, in)
+	return out, err
+}
+
+func (e *RPCEngine) ethGetBalance(
+	ctx context.Context,
+	retryCount int,
+	traceID string,
+	clock *pbsubstreams.Clock,
+	in []byte,
+) (outBytes []byte, deterministic bool, err error) {
+
+	reqMsg := &pbethss.RpcGetBalanceRequests{}
+	if err := proto.Unmarshal(in, reqMsg); err != nil {
+		return nil, false, fmt.Errorf("unmarshal get_balance proto: %w", err)
+	}
+
+	if len(reqMsg.Requests) == 0 {
+		return []byte{}, true, nil
+	}
+
+	rpcReqs := make([]*rpc.RPCRequest, len(reqMsg.Requests))
+	for i, r := range reqMsg.Requests {
+		addrHex := "0x" + hex.EncodeToString(r.Address)
+		rpcReqs[i] = &rpc.RPCRequest{
+			Method: "eth_getBalance",
+			Params: []interface{}{addrHex, clock.Id},
+		}
+	}
+
+	responses, det, err := e.rpcDoWithRetry(ctx, traceID, retryCount, clock.Id, rpcReqs)
+	if err != nil {
+		return nil, det, err
+	}
+
+	outMsg := &pbethss.RpcGetBalanceResponses{}
+	for _, resp := range responses.Responses {
+		outMsg.Responses = append(outMsg.Responses, &pbethss.RpcGetBalanceResponse{
+			Balance: resp.Balance,
+			Failed:  resp.Failed,
+		})
+	}
+	outBytes, err = proto.Marshal(outMsg)
+	return outBytes, det, err
+}
+
+// rpcGetBalanceWithRetry does exactly what rpcCalls does for eth_call, but for eth_getBalance. It returns a fully-populated *RpcGetBalanceResponses plus the deterministic flag.
+func (e *RPCEngine) rpcDoWithRetry(
```

Comment:
the name of this method makes it unclear that it is for getBalance, but it is!
e.rpcCalls() could be renamed e.ethCalls
this func here could be renamed e.getBalances since it is the "plural version" of e.getBalance

clarify the intent with naming that makes sense with other funcs